### ISSUE TITLE
CDS/CDNSKEY delete form is not handled correctly

### ISSUE THREAD

**Issue by ulrichwisser** [zonemaster/zonemaster-engine#964]
Today we did run checks on joinly.se. The domain is hosted on Cloudflare and tries to remove DNSSEC with the approriate cds/cdnskey records. 
Zonemaster reports that these records are not matching. Although all  cds records are of the form "... CDS 0 0 0 00" and cdnskey records are of the form  "... CDNSKEY 0 3 0 AA==" as demanded by the RFC.

**Comment by matsduf** [comment]
Strictly speaking, it does not seem like the records meet the RFC by letter.

```
;; ANSWER SECTION:
joinly.se.		3600	IN	CDS	0 0 0 00

;; ANSWER SECTION:
joinly.se.		3600	IN	CDNSKEY	0 3 0 AA==
```

```
   The contents of the CDS or CDNSKEY RRset MUST contain one RR and only
   contain the exact fields as shown below.

      CDS 0 0 0 0

      CDNSKEY 0 3 0 0

   The keying material payload is represented by a single 0.  This
   record is signed in the same way as regular CDS/CDNSKEY RRsets are
   signed.

   Strictly speaking, the CDS record could be "CDS X 0 X 0" as only the
   DNSKEY algorithm is what signals the DELETE operation, but for
   clarity, the "0 0 0 0" notation is mandated -- this is not a
   definition of DS digest algorithm 0.  The same argument applies to
   "CDNSKEY 0 3 0 0"; the value 3 in the second field is mandated by
   [RFC4034], Section 2.1.2.
```
(From [RFC 8078, section 4])

Is that domain under your control? Can you make the records fully RFC compliant?


[RFC 8078, section 4]: https://datatracker.ietf.org/doc/html/rfc8078#section-4

**Comment by matsduf** [comment]
Could be a bug.

**Comment by ulrichwisser** [comment]
No, we do not control the domain. There is an errata to to RFC8078 https://www.rfc-editor.org/errata_search.php?rfc=8078&rec_status=0 which changes the format to "0 3 0 AA==".

**Comment by vlevigneron** [comment]
DNSSEC16 and DNSSEC17 seems to give a correct result. The problem seems to be only with DNSSEC15. Here is the result obtained:

`17.29 ERROR     DNSSEC15       Both CDS and CDNSKEY RRsets are found on nameservers that resolve to IP addresses (108.162.192.135;108.162.193.143;172.64.32.135;172.64.33.143;173.245.58.135;173.245.59.143;2606:4700:50::adf5:3a87;2606:4700:58::adf5:3b8f;2803:f800:50::6ca2:c087;2803:f800:50::6ca2:c18f;2a06:98c1:50::ac40:2087;2a06:98c1:50::ac40:218f) but they do not match.`

I'm working on it.

**Comment by matsduf** [comment]
Only DNSSEC15 compares CDS and CDNSKEY.

**Comment by vlevigneron** [comment]
Yes, but RFC errata is on the format of both CDS and CDNSKEY for DELETE operation. So it could have on impact in DNSSEC16 and DNSSEC17 which do some test on these specific keys.

**Comment by matsduf** [comment]
Good point! 

Having standard in errata is strange. I would be better to update the RFC and call it revision 2.

**Comment by matsduf** [comment]
Resolved by #969.
